src/foam/box/DedicatedWorkerRegistry.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+foam.CLASS({
+  package: 'foam.box',
+  name: 'DedicatedWorkerRegistry',
+  extends: 'foam.box.BoxRegistryBox', //'foam.box.ProxyBox',
+  implements: [ 'foam.box.ProxyBox' ], //'foam.box.BoxRegistryBox' ],
+
+  requires: [
+    'foam.core.StubFactorySingleton',
+    'foam.box.BoxRegistry',
+    'foam.box.node.ForkBox'
+  ],
+
+//  imports: [ 'registry' ], // __context__.registry is the context it is declared in
+
+  exports: [ 'as registry' ], // __subContext__.registry === this
+
+  properties: [
+    {
+      name: "getDedicatedWorkerKey",
```

Comment:
`class: 'Function'`. You may need to use a `factory` instead of `value` if `foam.core.Function` has a factory.